modules/migrations/gitea_uploader.go

```diff
@@ -690,6 +690,23 @@ func (g *GiteaLocalUploader) newPullRequest(pr *base.PullRequest) (*models.PullR
 		}
 	} else {
 		head = pr.Head.Ref
+		// Ensure the closed PR SHA still points to an existing ref
+		_, err = git.NewCommand("rev-list", "--quiet", "-1", pr.Head.SHA).RunInDir(g.repo.RepoPath())
+		if err != nil {
+			if pr.Head.SHA != "" {
```

Comment:
```suggestion
			if len(pr.Head.SHA) != 0 {
```

Reply:
The funny thing is, as far as I have seen:
You are the only one who enforces this.
Almost everyone else uses `if str != ""`.

Reply:
yeah because it was quite some time ago decided to use `len(str) == 0` so I'm manually linting it 😆

Reply:
I also vote for `s != ""`, although there is no difference when the code gets compiled.

* `s != ""`
* `len(s) != 0`

The first one is much shorter, and easy to read.

Reply:
> yeah because it was quite some time ago decided to use `len(str) == 0` so I'm manually linting it laughing

Should we maybe have another vote on that?
I think that's the fastest way to resolve this, and then everyone knows how to handle this case in the future,

Reply:
I don't mind for any of options, ex golang stdlib also uses both we just need to keep to the one in our code base.

If I remember correctly len was decided to use because you don't have to think if it's string, byte array or rune array, it will work equally good for all cases. That affects also refactoring later on.

Reply:
The problem however is that as far as I have seen, both are used currently inside the code.
Sometimes `str != ""` is used, and sometimes (most likely when you reviewed the PR) `len(str) != 0` is used.

Reply:
> because you don't have to think if it's string, byte array or rune array, it will work equally good for all cases

I think this point is not that important. When we use a `string`, we do not new `s := make()`, we do not iterate `for x := s` , we do not update `s[k] = v`, and a `string` is immutable while array/map are mutable. They differ so much so we do not have to use `len` to all of them.

Reply:
I was not the one to propose it back in the days so don't shoot the messenger 😋 I was just pointing out that that's what was agreed on and when I review and see that I just point it out. I'm used to using it myself and for myself it's easier to use len than compare to empty string as I don't have to think so much on a type used. Sadly I can't seem to find where it was discussed as it was so long ago 😂